### PR TITLE
ci: support for CI on whitelisted PRs 

### PR DESCRIPTION
**Pull request by tazlin**
This PR will introduce the repository behavior of allowing the CI to run on PRs **if, and only if** the PR itself is labeled with the github label `allow-ci`.

Note that without this extra layer of protection, arbitrary code could potentially be run using the secrets in this repository. Presently, the CI only runs on push to main. This PR's approach still isn't quite perfect but so long as the label is only applied to known (trusted) parties' PRs, this should be safe enough.

### CODE REVIEW

**Comment by db0** [comment]
The PRing person can't add labels? I can't remember :D

**Comment by tazlin** [comment]
https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#applying-a-label

TLDR: you need triage access at least to a repo to apply a label
